### PR TITLE
[TESTS] Operations with data structures

### PR DESCRIPTION
**Pull request by Juan-M-V**
# Operations with data 

## Description

Added tests for `find_element`, `search_sorted_lower` and `usort`

## Checklist
- [x] Integration tests added.

### CODE REVIEW

File: cairo_programs/operations_with_data.cairo [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    let bike_arr:Bicycle* = alloc()
+    fill_bike_array(bike_arr, 30, 0)
+    let (element_ptr : Bicycle*) = find_element(
+        array_ptr=bike_arr,
+        elm_size=Bicycle.SIZE,
+        n_elms=30,
+        key=29,
+    )
+
+    assert element_ptr.gear_ratios.crankset = 0
```
Comment (entropidelic) [inline cairo_programs/operations_with_data.cairo:63]: Could these assertions be made with a function that checks the assertion element by element?

Comment (entropidelic) [reply cairo_programs/operations_with_data.cairo:63]: If this adds a lot of additional work don't worry

**Comment by entropidelic** [comment]
These test will be used to benchmark hints and compare the rust hints implementation vs the python interpreted ones, so it would be nice to have a way to iterate over the tests, just like we are doing with the other integration tests. For this you can check `math_integration_tests.cairo`, `memory_integration_tests.cairo` and `math_cmp_and_pow_integration_tests.cairo`

File: cairo_programs/operations_with_data.cairo (hunk 1 above)
Comment (Juan-M-V) [reply cairo_programs/operations_with_data.cairo]: I refactored the asserts where I could to be away from the main function. But I'm unsure if this is what you pointed out. https://github.com/lambdaclass/cleopatra_cairo/pull/362/commits/4452ab506a61988ddab8685f53d606719f2c5491

**Review by entropidelic — APPROVED** [review]

**Review by pefontana — APPROVED** [review]

**Pull request merged**